streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -740,19 +740,21 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
 
         final Map<Long, StreamThread.State> threadState = new HashMap<>(threads.length);
         final ArrayList<StateStoreProvider> storeProviders = new ArrayList<>();
-        for (int i = 0; i < threads.length; i++) {
+        for (int i = 0; i <= threads.length; i++) {
```

Comment:
This causes an ArrayIndexOutOfBoundsException:
`i <= threads.length`

Reply:
@raouldh Thank you! Fixed.